### ISSUE TITLE
gcc arm-none-eabi: stdlib not detected because --specs=nosys.specs is passed to the compiler twice

### ISSUE THREAD

**Issue by cl0rm** [microsoft/vscode-cpptools#9531]
OS: Ubuntu 20.04
stdlib is in /usr/lib/gcc/arm-none-eabi/9.2.1/include

makefile tool correctly processes my rather large makefiles, but does not include (parts of ?? ) the stdlib (which of course is not mentioned in the makefile -- the compiler automatically includes it).

This ends in intelliSense not finding stddef.h and stdbool.h

c/c++ log says: 

  Taganalysedatei: /usr/include/newlib/c++/9.2.1/tr1/stdbool.h
  Taganalysedatei: /usr/include/c++/9/tr1/stdbool.h
and
  Taganalysedatei: /usr/include/linux/stddef.h
both the wrong locations

it receives the following configuration:

```Benutzerdefinierte Suchkonfiguration empfangen: {
  "browsePath": [
    "/home/silbuntu/Repositories/adkt-stack/Applications/PAL_Examples/PAL_UART/Inc",
    "/home/silbuntu/Repositories/adkt-stack/Applications/PAL_Examples/PAL_UART/Src",
    "/home/silbuntu/Repositories/adkt-stack/Applications/TEST_UART_B/PAL_UART/Inc",
    "/home/silbuntu/Repositories/adkt-stack/Applications/TEST_UART_B/PAL_UART/Src",
    "/home/silbuntu/Repositories/adkt-stack/Include",
    "/home/silbuntu/Repositories/adkt-stack/PAL/Inc/",
    "/home/silbuntu/Repositories/adkt-stack/PAL/STM32/Generic/Inc",
    "/home/silbuntu/Repositories/adkt-stack/PAL/STM32/Generic/Src",
    "/home/silbuntu/Repositories/adkt-stack/PAL/STM32/STM32F4xx/Boards",
    "/home/silbuntu/Repositories/adkt-stack/PAL/STM32/STM32F4xx/Boards/NUCLEO_F429ZI",
    "/home/silbuntu/Repositories/adkt-stack/PAL/STM32/STM32F4xx/Inc",
    "/home/silbuntu/Repositories/adkt-stack/PAL/STM32/STM32F4xx/Inc/config",
    "/home/silbuntu/Repositories/adkt-stack/PAL/STM32/STM32F4xx/Src",
    "/home/silbuntu/Repositories/adkt-stack/PAL/STM32/STM32F4xx/Src/config"
  ],
  "compilerArgs": [
    "-c",
    "-mcpu=cortex-m4",
    "-march=armv7e-m",
    "--specs=nosys.specs",
    "-Wall",
    "-g",
    "-Wundef",
    "-Os",
    "-MD",
    "-MP",
    "-MT",
    "retarget_stdio.o",
    "--function-sections",
    "-mthumb",
    "-o",
    "obj/retarget_stdio.o",
    "../../../../../PAL/STM32/Generic/Src/retarget_stdio.c"
  ],
  "compilerPath": "/usr/bin/arm-none-eabi-gcc",
  "standard": "c99",
  "windowsSdkVersion": ""
}
```

Edit: found the culprit: 
C/C++ Tools executes the following:

```
/usr/bin/arm-none-eabi-gcc -mcpu=cortex-m4 -march=armv7e-m --specs=nosys.specs -Wall -g -Wundef -Os -MD -MP -MT main.o --function-sections -mthumb -mcpu=cortex-m4 -march=armv7e-m --specs=nosys.specs -Wall -g -Wundef -Os -MD -MP -MT main.o --function-sections -mthumb -std=c99 -Wp,-v -E -dM -x c /dev/null
```
then, the compiler throws a fatal error because  --specs=nosys.specs is included twice. But why is the section for main.o added twice? It is the same for every other file C/C++ Tools trys to process.

when omitting the --specs=nosys.specs parameter in the Makefile, it works perfectly fine and detects everything. It can only be included once in a compiler pass, so the following will also result in a correct output.
```
/usr/bin/arm-none-eabi-gcc -mcpu=cortex-m4 -march=armv7e-m --specs=nosys.specs -Wall -g -Wundef -Os -MD -MP -MT main.o --function-sections -mthumb -mcpu=cortex-m4 -march=armv7e-m -Wall -g -Wundef -Os -MD -MP -MT main.o --function-sections -mthumb -std=c99 -Wp,-v -E -dM -x c /dev/null
```
How to fix this issue properly? because omitting the argument is not a solution - it is very important for the build process. Without it, the linker will fail.

**Comment by Colengms** [comment]
Hi @cl0rm .  Thanks for reporting this.  I've transferred this to the cpptools repo.  I believe I've tracked down the issue and have a fix that should be available in our next insiders release.

**Comment by sean-mcmanus** [comment]
Fixed with 1.10.8 (or the latest pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.10.8
